File: AppReactNative/pages/menu.js
import React from "react";
import { View, Text, TouchableOpacity, StyleSheet } from "react-native";
import Icon from "react-native-vector-icons/MaterialIcons";

export default function menu({ navigation }) {

  return (
    <View style={styles.container}>

      <Text style={styles.title}>Titulo de pagina</Text>

      <Text style={styles.subtitle}>
        Selecciona un área para marcar tu cita:
      </Text>

      <View style={styles.grid}>

        <TouchableOpacity style={styles.card}>
          <Icon name="medical-services" size={50} color="#2a86d1"/>
          <Text style={styles.cardText}>Odontología</Text>
        </TouchableOpacity>

        <TouchableOpacity style={styles.card}>
          <Icon name="accessibility" size={50} color="#2a86d1"/>
          <Text style={styles.cardText}>Traumatología</Text>
        </TouchableOpacity>

        <TouchableOpacity style={styles.card}>
          <Icon name="psychology" size={50} color="#2a86d1"/>
          <Text style={styles.cardText}>Psicología</Text>
        </TouchableOpacity>

        <TouchableOpacity style={styles.card}>
          <Icon name="local-hospital" size={50} color="#2a86d1"/>
          <Text style={styles.cardText}>Rayos X</Text>
        </TouchableOpacity>

      </View>

    </View>
  );
}

const styles = StyleSheet.create({

  container:{
    flex:1,
    backgroundColor:"#e5e3e8",
    paddingTop:40,
    alignItems:"center"
  },

  title:{
    fontSize:20,
    marginBottom:10
  },

  subtitle:{
    fontSize:18,
    fontWeight:"bold",
    textAlign:"center",
    marginBottom:30
  },

  grid:{
    width:"90%",
    flexDirection:"row",
    flexWrap:"wrap",
    justifyContent:"space-between"
  },

  card:{
    width:"45%",
    height:140,
    backgroundColor:"#f2f2f2",
    borderRadius:20,
    justifyContent:"center",
    alignItems:"center",
    marginBottom:20,
    elevation:5
  },

  cardText:{
    marginTop:10,
    fontSize:16,
    fontWeight:"bold"
  }

});

File: AppFlutter/app_d/lib/pages/menu.dart
import 'package:app_d/pages/m_psicologia.dart';
import 'package:app_d/pages/m_rayox.dart';
import 'package:app_d/pages/m_traumatologia.dart';
import 'package:flutter/material.dart';
import 'm_odontologia.dart';


class menu extends StatelessWidget {
  const menu({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Titulo de pagina(Aun no tengoxd)"),
        centerTitle: true,
        backgroundColor: Colors.blue,
      ),
      body: Center(
        child: SizedBox(
          width: 350,
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [

              Text(
                "Selecciona un área para marcar tu cita:",
                textAlign: TextAlign.center,
                style: TextStyle(
                  fontSize: 20,
                  fontWeight: FontWeight.bold,
                ),
              ),

              SizedBox(height: 20),

              Expanded(
                child: GridView.count(
                  crossAxisCount: 2,
                  crossAxisSpacing: 20,
                  mainAxisSpacing: 20,
                  children: [

                    menuCard(
                      Icons.medical_services, "Odontología", () {
                        Navigator.push(
                          context,
                          MaterialPageRoute(
                            builder: (context) => m_odon()),
                        );
                      },
                    ),
                    menuCard(Icons.accessibility_new, "Traumatología", () {
                        Navigator.push(
                          context,
                          MaterialPageRoute(
                              builder: (context) => m_trauma()),
                        );
                      },
                    ),

                    menuCard(Icons.psychology, "Psicología", () {
                        Navigator.push(
                          context,
                          MaterialPageRoute(
                              builder: (context) => m_psicolog()),
                        );
                      },),



                    menuCard(Icons.local_hospital, "Rayos X", () {
                        Navigator.push(
                          context,
                          MaterialPageRoute(
                              builder: (context) => m_rayox()),
                        );
                      },),

                  ],
                ),
              ),

            ],
          ),
        ),
      ),

    );
  }

  Widget menuCard(IconData icon, String title, VoidCallback onTap) {
    return Card(
      elevation: 6,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(20),
      ),
      child: InkWell(
        onTap: onTap,
        borderRadius: BorderRadius.circular(20),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [

            Icon(
              icon,
              size: 60,
              color: Colors.blue,
            ),

            SizedBox(height: 10),

            Text(
              title,
              style: TextStyle(
                fontSize: 18,
                fontWeight: FontWeight.bold,
              ),
            )

          ],
        ),
      ),
    );
  }

}
